bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rntuple.py

```diff
@@ -24,14 +24,24 @@ def _REntry_GetPtr(self, key):
 
 
 def _REntry_getitem(self, key):
-    ptr = self.GetPtr(key)
-    return ptr.get()[0]
+    ptr_proxy = self.GetPtr(key)
+    # Most types held by a shared_ptr are exposed by cppyy as the type of the
+    # pointee, so we need to get the smart pointer before dereferencing. This
+    # does not happen for smart pointers to fundamental types.
+    real_smartptr = ptr_proxy.__smartptr__() if ptr_proxy.__smartptr__() is not None else ptr_proxy
```

Comment:
I think this should move to `_REntry_GetPtr`, we want this also for that lower level function.